### PR TITLE
ice: multiple formula fixes

### PR DESCRIPTION
**Pull request by externl**
- Add Java 8 support
- Always build slice2java and slice2freezej compilers
- Add test

### CODE REVIEW

File: Library/Formula/ice.rb [hunk 1]
```diff
@@ -89,4 +81,39 @@ def install
...
+        adapter->add(new HelloI, communicator->stringToIdentity("hello"));
+        adapter->activate();
+        communicator->destroy();
+        return 0;
+      }
+    EOS
+    system "#{bin}/slice2cpp", "Hello.ice"
+    system "xcrun", "clang++", "-c", "-I#{include}", "-I.", "Hello.cpp"
+    system "xcrun", "clang++", "-c", "-I#{include}", "-I.", "Test.cpp"
+    system "xcrun", "clang++", "-L#{lib}", "-o", "test", "Test.o", "Hello.o", "-lIce", "-lIceUtil"
```
Comment (mikemcquaid) [inline Library/Formula/ice.rb]: Does it fail with the `xcrun`?

Comment (externl) [inline Library/Formula/ice.rb]: xcrun worked fine for me locally.

Comment (mikemcquaid) [inline Library/Formula/ice.rb]: Sorry, I meant without the `xcrun`; don't think it should be necessary.

Comment (externl) [inline Library/Formula/ice.rb]: Fails to link. `xcrun` adds some necessary Frameworks automatically.

Comment (mikemcquaid) [inline Library/Formula/ice.rb]: Ok, thanks.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**